client/src/test/login.test.js

```diff
@@ -0,0 +1,48 @@
+// Login Form Full DOM Rendering Test
+import React from 'react';
+import Login from '../containers/Login/Login';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import store from '../store/configureStore';
+import { BrowserRouter as Router } from 'react-router-dom';
+import * as actions from '../actions';
+
+const WrappedComponent = () => {
+  return (
+    <Provider store={store}>
+      <Router>
+        <Login />
+      </Router>
+    </Provider>
+  );
+};
+
+describe('<Login />', () => {
+  const mockLogin = jest.spyOn(actions, 'userLogin');
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = mount(<WrappedComponent />);
+  });
+
+  it('Component should render without crashing', async () => {
+    expect(wrapper.find('form').length).toEqual(1);
+  });
+
+  it('Component should trigger login', () => {
```

Comment:
@shahnawaz19415 what about testing the validations

Reply:
Redux form validations?

Reply:
yes

Reply:
will add a check.

Reply:
Redux fields need to be tested in isolation. Separate tests are required to check if fields work properly or not.